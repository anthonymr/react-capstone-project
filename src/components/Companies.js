import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { PropTypes } from 'prop-types';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGear, faChevronLeft, faArrowRight } from '@fortawesome/free-solid-svg-icons';
import { fetchCompanies } from '../redux/companies/companiesSlice';
import style from '../styles/Companies.module.css';
import ICONS from '../assets/incons';

function Companies({ sector }) {
  const { companies } = useSelector((state) => state.companies);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCompanies(sector));
  }, [dispatch, sector]);

  return (
    <section>
      <header>
        <div>
          <NavLink to="/">
            <FontAwesomeIcon icon={faChevronLeft} />
          </NavLink>
        </div>
        <div>Stock Screaner</div>
        <div><FontAwesomeIcon icon={faGear} /></div>
      </header>
      <div className="header-container">
        <FontAwesomeIcon icon={ICONS[sector]} />
        <div>{sector}</div>
      </div>

      <section className={style.companiesContainer}>
        {
          companies.map((company) => {
            const url = encodeURI(`/company/${company.symbol}`);

            return (
              <NavLink to={url} key={company.symbol}>
                <div className={style.symbol}>{company.symbol}</div>
                <div className={style.info}>
                  <div>{company.companyName}</div>
                  <div>{company.price}</div>
                  <div>{company.beta}</div>
                </div>
                <div className={style.arrowRight}>
                  <FontAwesomeIcon icon={faArrowRight} />
                </div>

              </NavLink>
            );
          })
        }
      </section>
    </section>
  );
}

Companies.propTypes = {
  sector: PropTypes.string.isRequired,
};

export default Companies;
