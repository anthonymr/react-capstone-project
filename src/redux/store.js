import { configureStore } from '@reduxjs/toolkit';
import sectorsReducer from './sectors/sectorsSlice';

const store = configureStore({
  reducer: {
    sectors: sectorsReducer,
  },
});

export default store;
