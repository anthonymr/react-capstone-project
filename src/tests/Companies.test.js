import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/store';
import Companies from '../components/Companies';

const mockedSector = encodeURI('Basic Materials');

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <Companies sector={mockedSector} />
        </Provider>
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
