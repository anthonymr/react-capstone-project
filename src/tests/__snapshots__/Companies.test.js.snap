// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
>
  Home
</a>
`;
